pybandits/model.py

```
@@ -975,10 +1166,64 @@ def validate_activation(cls, v):
             )
         return v
 
+    @model_validator(mode="after")
+    def _infer_feature_config(self) -> Self:
+        """If feature_config is not provided, infer an all-numerical config from the first layer weight shape."""
+        if self.feature_config is None:
+            layers = self.model_params.bnn_layer_params
+            if not layers:
+                raise ValueError("feature_config is required when model_params.bnn_layer_params is empty.")
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)